src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -48,9 +48,11 @@ public class VsPackageInstaller : IVsPackageInstaller2
             _sourceRepositoryProvider = sourceRepositoryProvider;
             _settings = settings;
             _solutionManager = solutionManager;
-            _projectContext = new VSAPIProjectContext();
             _packageServices = packageServices;
             _deleteOnRestartManager = deleteOnRestartManager;
+
+            _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
```

Comment:
The long term fix sounds good.

I'm not sure the other change would be larger, it would just be making source control manager lazy in the project context, instead of making the project context lazy in the current class. The constructor for project context is minimal apart from the service locator call.